### ISSUE TITLE
adding a textTrack to the video

### ISSUE THREAD

**Issue by eladonline** [google/shaka-player#1401]
**Have you read the [FAQ](https://goo.gl/JE1Sy5) and checked for duplicate open issues?**:
YES
**What version of Shaka Player are you using?**:
5.6.0
**Can you reproduce the issue with our latest release version?**:
yes
**Can you reproduce the issue with the latest code from `master`?**:
dont understand whats the meaning
**Are you using the demo app or your own custom app?**:
my own custom app 
**If custom app, can you reproduce the issue using our demo app?**:
i cant use your demo up its blocking my lnk for security reasons
**What browser and OS are you using?**:
window 8.1 , chrome
**What are the manifest and license server URIs?**:
*(NOTE: you can send the URIs to <shaka-player-issues@google.com> instead, but please use GitHub and the template for the rest)*
this is the track that return when running player.getTextTracks();
```
      {id: 3, active: true, type: "text", bandwidth: 0, language: "he", …}
active:true
audioBandwidth:null
audioCodec:null
audioId:null
bandwidth:0
channelsCount:null
codecs:null
frameRate:null
height:null
id:3
kind:"subtitle"
label:null
language:"he"
mimeType:"text/vtt"
primary:false
roles:[]
type:"text"
videoBandwidth:null
videoCodec:null
videoId:null
width:null
__proto__:Object
```
*(NOTE: a copy of the manifest text or an attached manifest will **not** be enough to reproduce your issue, and we **will** ask you to send a URI instead)*

**What did you do?**
I  want to add subtitles to the  video so i did that 

```
    const letSource =
      'MyUrl.mpd';
    player
      .load(letSource) // this here load the channels it gets !!!
      .then(() => {
        // This runs if the asynchronous load is successful.
        console.log('The video has now been loaded!');

         const [track] = player.getTextTracks();
         player.selectTextTrack(track.id); // => this finds the id and everythings fine but it does nothing 
        const isBool = player.selectTextTrack(track.id);
        console.log(isBool); // => this returns undefined
      })
      .catch(this.onError); // onError is executed if the asynchronous load fails.

```

**What did you expect to happen?**
expected  isBool  to be a boolean
**What actually happened?**
isBool is undefined

**Comment by joeyparrish** [comment]
As you observed, selectTextTrack has no return value.  The documentation agrees with that: https://v2-3-5-dot-shaka-player-demo.appspot.com/docs/api/shaka.Player.html#selectTextTrack

Have you tried enabling the text track with setTextTrackVisibility?  https://v2-3-5-dot-shaka-player-demo.appspot.com/docs/api/shaka.Player.html#setTextTrackVisibility

**Comment by eladonline** [comment]
I have added  `track.setTextTrackVisibility(true)` to the line after `const [track] = player.getTextTracks();`
and still nothing I have  only the shaka default textTrack wich is playing nothing

**Comment by eladonline** [comment]
still waiting for reply

**Comment by joeyparrish** [comment]
@eladonline, JFYI, everyone on the team is on US/Pacific time, so you should not expect replies outside of normal office hours in our time zone.  When your own time zone is offset from ours by more than 6-8 hours, this introduces a lot of latency into communication.  That's just the nature of having a global conversation.  :-)

I think the problem is that it's not `track.setTextTrackVisibility(true)` as you wrote, but `player.setTextTrackVisibility(true)`.  Please compare your code to the docs I linked above, and also check the JavaScript console.  If you call that method on Track instead of Player, you should see an exception in the console telling you that there is no such method.

Does that help?

**Comment by eladonline** [comment]
joeyparrish Im sorry if you feel I was pressing you to answer it was not my intention
I just wanted to not be forgotten 

now to the problem 
I changed it to `player.setTextTrackVisibility(true)` 
it made the default doing nothing "shaka track" disappear, but still doesnt play subtitles or making a subtitle item in the  subtitle  menu

**Comment by joeyparrish** [comment]
No worries.

If it works in JWPlayer and not in your custom app, the next question is whether or not it works in our demo app.  Can you please try in our demo?  You can select "custom asset" from the assets list and enter a URL.  You will need to either allow cross-origin access to your asset or disable CORS restrictions in your browser.

I would also like you to confirm what version of Shaka Player you are using.  Your report says "5.6.0", but that version doesn't exist.  Our latest release is v2.3.6.  What version are you using?

**Comment by eladonline** [comment]
hey sorry fro the wrong information 
the currently use version is 2.3.4 
(BTW I  updated to 2.3.6 and it threw a lot of errors so I downgrade it again)
I open my chrome with this command
`chrome.exe --user-data-dir="C:/Chrome dev session" --disable-web-security`
and when trying to run the mpd url on your 'shaka demo' it throws an err 1002 
![image](https://user-images.githubusercontent.com/20336964/38733649-636a9412-3f2b-11e8-8829-c3951cc71b1b.png)
if you want the url of the mpd, I sent it to your supp email.
the head of the email holds this issue id track #1401
so you can try it too

**Comment by joeyparrish** [comment]
I found your email, but it does not contain a manifest URI.  It appears to be just your email signature and no content.  Can you please send another email with the manifest URI?

As for HTTP_ERROR, that still appears to be CORS at work.  Due to a suspected Chrome bug, to disable CORS, you must use both --disable-web-security and open the debugger, go to the Application tab, click the "Service worker" section, and check the "Bypass for network" box.  Please do that and report your results.

Thanks!

**Comment by eladonline** [comment]
@joeyparrish hey did you find something?

**Comment by joeyparrish** [comment]
@eladonline, I'm sorry, we haven't looked at this yet.  Thank you for your patience.

**Comment by ismena** [comment]
Hi @eladonline, I took a look at your manifest and got an error when trying to display subtitles. I'll look into it later this week.
The original problem was that the subtitles didn't display on your custom app, is that right?

**Comment by ismena** [comment]
Got it. I should have an update for you by the end of the week.

**Comment by ismena** [comment]
Ok, looks like there is a problem with your subtitle file which prevents us from parsing it.
Here is the snippet from it:
3
00:00:07.000 --> 00:00:09.000
הרוכבים הפלמים 

4
00:00:09.000 --> 00:00:11.000


5
00:00:11.000 --> 00:00:13.000




















טסט קומיטד

It's not obvious with the comment formatting, but there are a bunch of newlines between cue #5 and it's payload (about 20 newlines in the actual file :)).
According to the [VTT spec](https://www.w3.org/TR/webvtt1/#webvtt-cue-block) cue timimg should be directly followed with one newline character and then the text payload.
There is another spot like this in your vtt file: between cue #17 and it's text.
Once you fix the content everything should work :)

**Comment by eladonline** [comment]
@ismena I did as you asked and it works issue can be close, tvm!
Need to mention that if One use shaka player with big companies 
it is impossible to stand in the translation criteria
because every company use her tool to make translates and some make more spaces 
this propblem is only in shaka player other players work well with this 
maybe this issue should turn into a suggestion

**Comment by ismena** [comment]
@eladonline glad it worked! Let me asked our TL about supporting these kinds of files.

@joeyparrish 
Tl;dr: apparently some tools used for translating vtt subtitles add more spaces between timing and payload than the spec allows. We can't parse them. Are we willing to modify the VttParser to be able to recover from this?
From code perspective, it's not a gigantic change, but not a small one either since a lot of our logic was built around the assumption that 2 or more new lines indicate a new text cue.

**Comment by joeyparrish** [comment]
@eladonline, if you can provide us with samples and the tools that created them, we may be able to support these non-standard VTT files.  I would need to understand how they are created before we change the parser to accommodate.  Please file a new issue if you want to pursue this.  Thanks!
